src/cares_wrap.cc

```diff
@@ -328,30 +450,91 @@ class QueryWrap : public AsyncWrap {
     return static_cast<void*>(this);
   }
 
-  static void Callback(void *arg, int status, int timeouts,
-      unsigned char* answer_buf, int answer_len) {
-    QueryWrap* wrap = static_cast<QueryWrap*>(arg);
+  static void CaresAsyncClose(uv_handle_t* handle) {
+    uv_async_t* async = reinterpret_cast<uv_async_t*>(handle);
+    struct CaresAsyncData* data =
+        static_cast<struct CaresAsyncData*>(async->data);
```

Comment:
I think in this case it should be fine to just use `auto data =`, the type is obvious from the `static_cast`